ArduCopter/takeoff.cpp

```diff
@@ -50,11 +50,20 @@ bool Copter::Mode::do_user_takeoff(float takeoff_alt_cm, bool must_navigate)
 // start takeoff to specified altitude above home in centimeters
 void Copter::Mode::_TakeOff::start(float alt_cm)
 {
+    if (running()) {
+        return;
+    }
+
+    // indicate we are taking off
+    copter.set_land_complete(false);
+    // tell position controller to reset alt target and reset I terms
+    copter.pos_control->init_takeoff();
```

Comment:
Actually, moving it to *after* those early-returns would make it a functional change!

The code that's being removed unconditionally ran these lines - since `start` is a void function they really didn't have much choice in the matter.

While I do agree that this all looks a little odd, this could actually be critical given we're resetting i-terms and the like.

Reply:
> While I do agree that this all looks a little odd, this could actually be critical given we're resetting i-terms and the like.

I don't understand this, how is it critical if the takeoff isn't running? It should be as critical as before takeoff is engaged...